test/api_container_create_test.go

```diff
@@ -231,3 +234,48 @@ func (suite *APIContainerCreateSuite) TestCreateNvidiaConfig(c *check.C) {
 
 	DelContainerForceMultyTime(c, cname)
 }
+
+func (suite *APIContainerCreateSuite) TestCreateWithMacAddress(c *check.C) {
+	cname := "TestCreateWithMacAddress"
+	mac := "00:16:3e:02:00:b7"
+	q := url.Values{}
+	q.Add("name", cname)
+	query := request.WithQuery(q)
+
+	obj := map[string]interface{}{
+		"Entrypoint":  []string{"top"},
+		"Image":       busyboxImage,
+		"MacAddress":  mac,
+		"NetworkMode": "bridge",
+	}
+	body := request.WithJSONBody(obj)
+
+	resp, err := request.Post("/containers/create", query, body)
+	c.Assert(err, check.IsNil)
+	CheckRespStatus(c, resp, 201)
+
+	q = url.Values{}
+	q.Add("detachKeys", "EOF")
+	query = request.WithQuery(q)
+
+	resp, err = request.Post("/containers/"+cname+"/start", query)
+	c.Assert(err, check.IsNil)
+	CheckRespStatus(c, resp, 204)
+
+	ret := command.PouchRun("exec", cname, "ip", "addr", "show", "eth0")
+	ret.Assert(c, icmd.Success)
+
+	found := false
+	for _, line := range strings.Split(ret.Stdout(), "\n") {
+		if strings.Contains(line, "link/ether") {
+			if mac == strings.Fields(line)[1] {
+				found = true
+				break
+			}
+		}
+	}
+
+	c.Assert(found, check.Equals, true)
+
+	DelContainerForceMultyTime(c, cname)
```

Comment:
We should use `defer` to delete container to avoid legacy container for other tests.